syntax = "proto3";

// Core
message Actor {}
message Interface {}
message Command {}
message Message {}

// Drone
message Drone {
  Actor systemBase = 1;
  TelemetryInterface telemetry = 2;
  ControllerInterface controller = 3;
}

// Drone - Telemetry
message TelemetryInterface {
  Interface systemBase = 1;
  RequestTelemetry requestTelemetry = 2;
  TelemetryUpdateMessage telemetryUpdateMessage = 3;
}

message RequestTelemetry {
  Command systemBase = 1;
  float updatePeriod = 2;
}

message TelemetryUpdateMessage {
  Message systemBase = 1;
  float batteryLevel = 2;
}

// Drone - Controller
message ControllerInterface {
  Interface systemBase = 1;
  TakeOff takeOff = 2;
}

message TakeOff {
  Command systemBase = 1;
}
